src/test/unit/services/util/initialize_test.cpp

```diff
@@ -256,15 +257,53 @@ namespace test {
   };
 }
 
-TEST_F(ServicesUtilInitialize, model_throws) {
+TEST_F(ServicesUtilInitialize, model_throws__radius_zero) {
+  std::vector<double> params;
+  test::mock_throwing_model throwing_model;
+
+  double init_radius = 0;
+  bool print_timing = false;
+  EXPECT_THROW(params = stan::services::util::initialize(throwing_model, empty_context, rng,
+                                                         init_radius, print_timing,
+                                                         message, init),
+               std::domain_error);
+
+  EXPECT_EQ(1, count_matches("throwing within log_prob", message_ss.str()));
+}
+
+TEST_F(ServicesUtilInitialize, model_throws__radius_two) {
   std::vector<double> params;
   test::mock_throwing_model throwing_model;
-    
+
+  double init_radius = 2;
+  bool print_timing = false;
+  EXPECT_THROW(params = stan::services::util::initialize(throwing_model, empty_context, rng,
+                                                         init_radius, print_timing,
+                                                         message, init),
+               std::domain_error);
+  EXPECT_EQ(100, count_matches("throwing within log_prob", message_ss.str()));
+}
+
+TEST_F(ServicesUtilInitialize, model_throws__full_init) {
+  std::vector<std::string> names_r;
+  std::vector<double> values_r;
+  std::vector<std::vector<size_t> > dim_r;
+  names_r.push_back("y");
+  values_r.push_back(6.35149);   // 1.5 unconstrained: -10 + 20 * inv.logit(1.5)
+  values_r.push_back(-2.449187); // -0.5 unconstrained
+  std::vector<size_t> d;
+  d.push_back(2);
+  dim_r.push_back(d);
+  stan::io::array_var_context init_context(names_r, values_r, dim_r);
+
+  std::vector<double> params;
+  test::mock_throwing_model throwing_model;
+
   double init_radius = 2;
   bool print_timing = false;
-  EXPECT_THROW(params
-               = stan::services::util::initialize(throwing_model, empty_context, rng,
-                                                  init_radius, print_timing,
-                                                  message, init),
+  EXPECT_THROW(params = stan::services::util::initialize(throwing_model, init_context, rng,
```

Comment:
Out of curiosity, why do you have to assign params to the result here? Is it so the compiler doesn't optimize the call away?